Makefile

```diff
@@ -15,8 +18,19 @@ GOLDFLAGS += -X github.com/algorand/indexer/version.ReleaseVersion=$(shell cat .
 export GO_IMAGE = golang:$(shell go version | cut -d ' ' -f 3 | tail -c +3 )
 
 # This is the default target, build the indexer:
-cmd/algorand-indexer/algorand-indexer:	idb/postgres/setup_postgres_sql.go idb/postgres/reset_sql.go types/protocols_json.go
-	cd cmd/algorand-indexer && CGO_ENABLED=0 go build -ldflags="${GOLDFLAGS}"
+cmd/algorand-indexer/algorand-indexer: idb/postgres/setup_postgres_sql.go idb/postgres/reset_sql.go types/protocols_json.go go-algorand
+	cd cmd/algorand-indexer && go build -ldflags="${GOLDFLAGS}"
+
+go-algorand-dl:
+	@if test -d ${GO_ALGORAND_CACHE_PATH}; then \
+		echo "go-algorand is already downloaded"; \
+	else \
+		echo "downloading go-algorand"; \
+		go get -d ${GO_ALGORAND_REPO}; \
+	fi
+
+go-algorand: go-algorand-dl
+	cd ${GO_ALGORAND_CACHE_PATH} && chmod +x `find -name *.sh` && chmod -R +w . && make
```

Comment:
Is this because `go get` is setting the wrong permissions?

Maybe we can avoid this by using a nested repository, or download with `git clone`?

Reply:
Yes, `go get` unsets those flags, probably for safety reasons. I will see what a nested repo will look like.